Makefile.inc1

```diff
@@ -578,7 +595,15 @@ BMAKEENV=	INSTALL="sh ${.CURDIR}/tools/install.sh" \
 		TOOLS_PREFIX=${WORLDTMP} \
 		PATH=${BPATH}:${PATH} \
 		WORLDTMP=${WORLDTMP} \
-		MAKEFLAGS="-m ${.CURDIR}/tools/build/mk ${.MAKEFLAGS}"
+		MAKEFLAGS="-m ${.CURDIR}/tools/build/mk ${.MAKEFLAGS}" \
+		COMPILER_TYPE="${COMPILER_TYPE}" \
```

Comment:
Does this DTRT if you aren't using an external compiler?  That is, when are building the in-tree clang, we would want these variables to reflect the compiler we are using to build that, but then when we build the actual world we want it to reflect the 'cc' we are using to build the world itself.

Reply:
This is for the bootstap-tools/cross-tools phase which should just be using the same compiler as the one that just got checked in bad.compiler.mk.

I'm not sure this change will work if you bootstrap clang but since we don't do that for cheribsd I didn't test it. I'll try to clean this up and submit upstream soon.